File: python programs/Dice.py
#!/usr/bin/python

import random


udice = input("How many sides on the die? ")
pdice = udice + 1
modifier = input("What is the bonus/negative modifier? ")
uroll = random.randrange(1, pdice)
froll = uroll + modifier

print "-" * 15
print "The modifier is", modifier
print "your roll is", uroll
print "-" * 15
print "your modified roll is", froll
print "-" * 15

File: perl programs/Dice.pl
#!/usr/bin/perl

use warnings;
use strict;

print "\n\n=======Dice Roller======\n\n";
print "How many sides on the die?\n";
my $sides = <>;
chomp($sides);
print "What is the modifier?\n";
my $mod = <>;
chomp($mod);
print "========================\n\n";


my ($roll, $mroll) = &droll($sides, $mod);
		

print "Your roll is $roll\n";
print "With a modifier of $mod\n";
print "========================\n\n";
print "Your final roll is $mroll\n\n";


sub droll {
	my $fsides = shift;
	my $fmod = shift;
	my $roll = int(rand($fsides));
	my $mroll = $roll + $fmod;
	return ($roll, $mroll);
}
